sky/skylet/providers/gcp/config.py

```diff
@@ -622,9 +624,30 @@ def _configure_key_pair(config, compute):
     return config
 
 
+def _get_required_rules(config):
+    """Returns the required firewall rules."""
+    default_rules = FIREWALL_RULES_REQUIRED.copy()
+    ports = config["provider"].get("ports", [])
+    for port in ports:
+        # Not set targetTags here to avoid filtered by _merge_and_refine_rule
```

Comment:
A much simpler way is that we don't compare the user-specified ports with the existing rules in the VPC, as those ports' lifecycle will be the same as the cluster.

We can keep the original check for `FIREWALL_RULES_REQUIRED` in the VPC, and have another codepath for creating firewall rules for the user-specified ports with `targetTags`.

Reply:
Done